core/core/ledgerkeeper.go

```diff
@@ -0,0 +1,1197 @@
+		return e.Value.(*LedgerTask)
+	}
+	return nil
+}
+
+func (stm *SyncTaskManager) fixTask(targetHeight int64) {
+	stm.syncMutex.Lock()
+	defer stm.syncMutex.Unlock()
+	stm.syncingTasks.fix(targetHeight)
+	stm.appendingTasks.fix(targetHeight)
+}
+
+type LedgerKeeper struct {
+	P2pSvr              p2p_base.P2PServer
+	log                 log.Logger
+	syncMsgChan         chan *xuper_p2p.XuperMessage
+	peersStatusMap      *sync.Map        // *map[string]bool 更新同步节点的p2p列表活性
+	fastFetchBlockCache *common.LRUCache // block header cache, 加速fetchBlock
+	maxBlocksMsgSize    int64            // 取最大区块大小
+	ledger              *ledger.Ledger
+	bcName              string
+	keeperStatus        KeeperStatus // 当前状态
+	ledgerMutex         *sync.Mutex  // 锁, Sync(异步)和Truncate(同步)抢锁
+	syncTaskMg          *SyncTaskManager
+	nodeMode            string
+
+	utxovm *utxo.UtxoVM
+	con    *consensus.PluggableConsensus
+}
+
+type LedgerTask struct {
+	taskId        string
+	action        KeeperStatus
+	targetBlockId string
+	targetHeight  int64
+	ctx           *LedgerTaskContext
+}
+
+type LedgerTaskContext struct {
+	extBlocks  *map[string]*SimpleBlock
+	preferPeer *[]string
+	hd         *global.XContext
+}
+
+func NewLedgerTaskContext(extBlocks *map[string]*SimpleBlock, preferPeer *[]string, hd *global.XContext) *LedgerTaskContext {
+	return &LedgerTaskContext{
+		extBlocks:  extBlocks,
+		preferPeer: preferPeer,
+		hd:         hd,
+	}
+}
+
+type SimpleBlock struct {
+	internalBlock *pb.InternalBlock
+	header        *pb.Header
+}
+
+func (lt *LedgerTask) GetTargetBlockId() string {
+	return lt.targetBlockId
+}
+
+func (lt *LedgerTask) GetAction() KeeperStatus {
+	return lt.action
+}
+
+func (lt *LedgerTask) GetHeight() int64 {
+	return lt.targetHeight
+}
+
+func (lt *LedgerTask) GetExtBlocks() map[string]*SimpleBlock {
+	if lt.ctx == nil || lt.ctx.extBlocks == nil {
+		return nil
+	}
+	return *lt.ctx.extBlocks
+}
+
+func (lt *LedgerTask) GetPreferPeer() []string {
+	if lt.ctx == nil || lt.ctx.preferPeer == nil {
+		return nil
+	}
+	return *lt.ctx.preferPeer
+}
+
+func (lt *LedgerTask) GetXCtx() *global.XContext {
+	if lt.ctx == nil || lt.ctx.hd == nil {
+		return nil
+	}
+	return lt.ctx.hd
+}
+
+func (lk *LedgerKeeper) PutTask(ledgerTask *LedgerTask) bool {
+	return lk.syncTaskMg.put(ledgerTask)
+}
+
+func (lk *LedgerKeeper) getTask() (*LedgerTask, bool) {
+	lk.ledgerMutex.Lock()
+	defer lk.ledgerMutex.Unlock()
+	// 刚刚完成Truncate操作，需要清除列表中已无效的task
+	if lk.keeperStatus == Truncateing {
+		lk.syncTaskMg.fixTask(lk.ledger.GetMeta().GetTrunkHeight())
+	}
+	task := lk.syncTaskMg.get()
+	if task == nil {
+		lk.keeperStatus = Waiting
+		return nil, false
+	}
+	lk.keeperStatus = task.GetAction()
+	return task, true
+}
+
+func NewLedgerKeeper(bcName string, slog log.Logger, p2pV2 p2p_base.P2PServer, maxBlocksMsgSize int64, ledger *ledger.Ledger, nodeMode string, utxovm *utxo.UtxoVM, con *consensus.PluggableConsensus) *LedgerKeeper {
```

Comment:
Done, now NewLedgerKeeper still has parameters over 3 because it includes initialization logic. Also, this way can make xchaincore's initialization more concise.